block/manager.go

```diff
@@ -64,6 +64,7 @@ type Manager struct {
 	// It is ALSO used by the producer, because the producer needs to check if it can prune blocks and it wont'
 	// prune anything that might be submitted in the future. Therefore, it must be atomic.
 	LastSubmittedHeight atomic.Uint64
+	TargetHeight        atomic.Uint64
```

Comment:
I think it's redundent. both `LastSubmittedHeight` and `TargetHeight` are actually `LatestHeightOnSL`